ophyd/pseudopos.py

```diff
@@ -227,22 +246,34 @@ def connected(self):
         return all(mtr.connected for mtr in self._real)
 
     def stop(self):
-        for pos in self._real:
-            pos.stop()
-
-        Positioner.stop(self)
+        try:
+            del self._move_queue[:]
+            for pos in self._real:
+                pos.stop()
+        finally:
+            super().stop(self)
 
-    def check_single(self, idx, single_pos):
+    def check_single(self, pseudo_single, single_pos):
         '''Check if a new position for a single pseudo positioner is valid'''
+        idx = pseudo_single._idx
         target = list(self.target)
         target[idx] = single_pos
-        return self.check_value(self.PseudoPosition(target))
+        return self.check_value(self.PseudoPosition(*target))
 
     def check_value(self, pseudo_pos):
         '''Check if a new position for all pseudo positioners is valid'''
-        pseudo_pos = self.PseudoPosition(pseudo_pos)
+        try:
+            pseudo_pos = self.PseudoPosition(*pseudo_pos)
+        except TypeError as ex:
+            raise ValueError('Not all required values for a PseudoPosition: {}'
+                             '({})'.format(self.PseudoPosition._fields, ex))
+
         for pseudo, pos in zip(self._pseudo, pseudo_pos):
-            pseudo.check_value(pos)
+            low, high = pseudo.limits
```

Comment:
Added a note about how we can't really support coupled limits generically - that sort of logic should be handled in subclasses when the default implementation doesn't apply.